lib/cretonne/src/licm.rs

```diff
@@ -16,9 +16,13 @@ pub fn do_licm(
     domtree: &mut DominatorTree,
     loop_analysis: &mut LoopAnalysis,
 ) {
-    cfg.ensure(func);
-    domtree.ensure(func, cfg);
-    loop_analysis.ensure(func, cfg, domtree);
+    debug_assert!(cfg.is_valid(), "LICM requires the CFG to be computed");
+    debug_assert!(domtree.is_valid(), "LICM requires a computed DominatorTree");
+    debug_assert!(
+        loop_analysis.is_valid(),
+        "LICM requires a computed LoopAnalysis"
+    );
```

Comment:
I think it would be ok to drop the explanatory text so these assertions don't go multi-line. The expression being asserted is quite self-explanatory.